dat/generated_tables.py

```diff
@@ -391,15 +393,149 @@ def create_stats_as_struct(case: TestCaseInfo, spark: SparkSession):
 def create_no_stats(case: TestCaseInfo, spark: SparkSession):
     df = get_sample_data(spark)
     (DeltaTable.create(spark)
-        .location(str(Path(case.delta_root).absolute()))
-        .addColumns(df.schema)
-        .property('delta.checkpointInterval', '2')
-        .property('delta.checkpoint.writeStatsAsStruct', 'false')
-        .property('delta.checkpoint.writeStatsAsJson', 'false')
-        .property('delta.dataSkippingNumIndexedCols', '0')
-        .execute())
+     .location(str(Path(case.delta_root).absolute()))
+     .addColumns(df.schema)
+     .property('delta.checkpointInterval', '2')
+     .property('delta.checkpoint.writeStatsAsStruct', 'false')
+     .property('delta.checkpoint.writeStatsAsJson', 'false')
+     .property('delta.dataSkippingNumIndexedCols', '0')
+     .execute())
 
     for i in range(3):
         df = get_sample_data(spark, seed=i, nrows=5)
         df.repartition(1).write.format('delta').mode(
             'overwrite').save(case.delta_root)
+
+
+@reference_table(
+    name='deletion_vectors',
+    description='Table with deletion vectors',
+)
+def create_deletion_vectors(case: TestCaseInfo, spark: SparkSession):
+    df = get_sample_data(spark)
+
+    delta_path = str(Path(case.delta_root).absolute())
+    delta_table: DeltaTable = (DeltaTable.create(spark)
+                               .location(delta_path)
+                               .addColumns(df.schema)
+                               .property('delta.enableDeletionVectors', 'true')
+                               .execute())
+
+    df.repartition(1).write.format('delta').mode('append').save(case.delta_root)
+
+    delta_table.delete(col("letter") == "a")
+
+
+@reference_table(
+    name='check_constraints',
+    description='Table with a check constraint'
+)
+def check_constraint_table(case: TestCaseInfo, spark: SparkSession):
+    df = get_sample_data(spark)
+    delta_path = str(Path(case.delta_root).absolute())
+    (DeltaTable.create(spark)
+     .location(delta_path)
+     .addColumns(df.schema)
+     .property('delta.enableDeletionVectors', 'true')
+     .execute())
+
+    df.repartition(1).write.format('delta').mode('append').save(case.delta_root)
+    spark.sql(f"ALTER TABLE delta.`{delta_path}` ADD CONSTRAINT const1 CHECK (int > 0);")
```

Comment:
Oh my, this syntax.

Reply:
this is the only way to add a constraint :-) delta-rs though has it part of the builder methods :-D